test/core/api/service/state/state_api_test.cpp

```diff
@@ -322,4 +324,106 @@ namespace kagome::api {
     EXPECT_OUTCOME_TRUE(result, get_metadata->execute());
     ASSERT_EQ(result, data);
   }
+
+  MATCHER_P(ContainedIn, container, "") {
+    return ::testing::ExplainMatchResult(
+        ::testing::Contains(arg), container, result_listener);
+  }
+
+  TEST_F(StateApiTest, QueryStorageSucceeds) {
+    std::vector<common::Buffer> keys{"key1"_buf, "key2"_buf, "key3"_buf};
+    primitives::BlockHash from{"from"_hash256};
+    primitives::BlockHash to{"to"_hash256};
+
+    std::vector block_range{from, "block2"_hash256, "block3"_hash256, to};
+    EXPECT_CALL(*block_tree_, getChainByBlocks(from, to))
+        .WillOnce(testing::Return(block_range));
+    EXPECT_CALL(*block_header_repo_,
+                getNumberByHash(from))
+        .WillOnce(testing::Return(1));
+    EXPECT_CALL(*block_header_repo_,
+                getNumberByHash(to))
+        .WillOnce(testing::Return(4));
+    for (auto &block_hash : block_range) {
+      primitives::BlockHash state_root;
+      auto s = block_hash.toString() + "_etats";
+      std::copy_if(s.begin(), s.end(), state_root.begin(), [](auto b) {
+        return b != 0;
```

Comment:
Not sure if **std::string** contains null-terminated string (C-string). You can use **std::string::c_str()** if it needed

Reply:
It's not about null terminator, it's to skip zeroes which ""_hash256 fills the empty space with.